modules/andes-core/broker/src/main/java/org/wso2/andes/store/file/FileMessageStoreImpl.java

```diff
@@ -0,0 +1,1364 @@
+    public DurableStoreConnection initializeMessageStore(AndesContextStore contextStore, ConfigurationProperties
+            connectionProperties) throws AndesException {
+        FileStoreConnection fileStoreConnection = new FileStoreConnection();
+        fileStoreConnection.initialize(connectionProperties);
+        this.brokerStore = (DB) fileStoreConnection.getConnection();
+        return fileStoreConnection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void storeMessages(List<AndesMessage> messageList) throws AndesException {
+        Transaction tx = new Transaction(brokerStore);
+
+        try {
+            for (AndesMessage message : messageList) {
+                storeMessage(message, tx);
+            }
+            tx.commit(brokerStore);
+        } catch (DBException e) {
+            log.error("Messages storing failed", e);
+        } finally {
+            tx.close();
+        }
+    }
+
+    /**
+     * add metadata, expire data and message content to the transaction
+     *
+     * @param message     message which needed to be stored
+     * @param transaction transaction which accumulates operations to commit
+     * @throws AndesException
+     */
+    private void storeMessage(AndesMessage message, Transaction transaction) throws AndesException {
+        AndesMessageMetadata metadata = message.getMetadata();
+
+        addMetadata(metadata, metadata.getStorageQueueName(), transaction);
+
+        if (metadata.isExpirationDefined()) {
+            addExpiryData(metadata, transaction);
+        }
+
+        for (AndesMessagePart messagePart : message.getContentChunkList()) {
+            addContent(messagePart, transaction);
+        }
+    }
+
+    /**
+     * add metadata details to the transaction
+     *
+     * @param metadata    message metadata which needed to be stored
+     * @param queueName   queue/topic name where the message is needed to be stored
+     * @param transaction transaction which accumulates operations to commit
+     *                    <p>
+     *                    Following keys will be created
+     *                    <p>
+     *                    MESSAGE.$message_id.DESTINATION_ID
+     *                    MESSAGE.$message_id.DLC_QUEUE_ID
+     *                    MESSAGE.$message_id.MESSAGE_METADATA
+     */
+    private void addMetadata(AndesMessageMetadata metadata, final String queueName, Transaction transaction)
+            throws AndesException {
+        String queueIDIdentifier = generateKey(DESTINATION_ID, MESSAGE, Long.toString(metadata.getMessageID()));
+        String dlcQueueIDIdentifier = generateKey(DLC_QUEUE_ID, MESSAGE, Long.toString(metadata.getMessageID()));
+        String messageMetaDataIdentifier = generateKey(MESSAGE_METADATA, MESSAGE, Long.toString(metadata.getMessageID()));
+
+        transaction.put(bytes(queueIDIdentifier), bytes(getDestinationID(queueName)));
+        transaction.put(bytes(dlcQueueIDIdentifier), bytes("-1"));
```

Comment:
Shall we define this "-1" as a constant?

Reply:
Fixed